var/spack/repos/builtin/packages/py-notebook/package.py

```diff
@@ -34,7 +34,7 @@ class PyNotebook(PythonPackage):
     depends_on('py-nbformat', type=('build', 'run'))
     depends_on('py-nbconvert', type=('build', 'run'))
     depends_on('py-ipykernel', type=('build', 'run'))
-    depends_on('py-ipykernel@5.1.0:', when='@4.2.0:', type=('build', 'run'))
+    depends_on('py-ipykernel@5.1.0:', when='@4.2.0: ^python@3:', type=('build', 'run'))
```

Comment:
This is required for both Python 2 and 3

Reply:
As far as I can tell, no. E.g., for `notebook==5.7.8` pip installs `ipykernel-4.10.1`:
```console
$ mktmpenv -p python2
(...)
Collecting ipykernel
  Downloading https://files.pythonhosted.org/packages/59/9e/e16335ee2d645ee48f082e4207d4fad9bdce09cc1537e76320c341d8d75c/ipykernel-4.10.1-py2-none-any.whl (109kB)
(...)
```

Reply:
Actual requirements for `notebook==5.7.8`:
```
Requires-Dist: jinja2
Requires-Dist: tornado (<7,>=4.1)
Requires-Dist: pyzmq (>=17)
Requires-Dist: ipython-genutils
Requires-Dist: traitlets (>=4.2.1)
Requires-Dist: jupyter-core (>=4.4.0)
Requires-Dist: jupyter-client (>=5.2.0)
Requires-Dist: nbformat
Requires-Dist: nbconvert
Requires-Dist: ipykernel
Requires-Dist: Send2Trash
Requires-Dist: terminado (>=0.8.1)
Requires-Dist: prometheus-client
Requires-Dist: ipaddress ; python_version == "2.7"
```